30_Day_Leetcode_Challenge/Week 3/Day15_ProductOfArrayExceptSelf.py

```diff
@@ -0,0 +1,25 @@
+class Solution:
+    def productExceptSelf(self, nums: List[int]) -> List[int]:
+        """Calculate the product of other elements of the array, except itself.
+
+        Args:
+          nums: the input array.
+
+        Returns:
+          An array output such that output[i] is equal to
+          the product of all the elements of nums except nums[i].
+          
+        """
+        result = [1]
+        current = 1
+        
+        for i in range(len(nums)-1): 
+            current *= nums[i]
+            result.append(current)
+            
+        current = 1
+        for i in range(len(nums)-1, -1, -1): 
+            result[i] *= current
+            current *= nums[i]
+            
```

Comment:
this solution is correct, only thing I would add is the asymptotic analysis on space and time, either in the code or in the Readme to explain your analysis, its a good habit to do that before we code (or after if the algo is complex)